### ISSUE TITLE
site.getsitepackages returns also the venv dir

### ISSUE THREAD

**Issue by timendum** [python/cpython#102733]
# Bug report

On Windows, the method `site.getsitepackages()` returns `<venv>\lib\site-packages` which is expected, but also `<venv>`, which I think is an issue.

From the documentation, it should "Return a list containing all global site-packages directories."

I've also checked in a Linux virtualenv environment, and it returns only "site-packages" and "dist-packages" folders.

# Demo

```console
PS> python -m venv test
PS> cd .\test\
PS> .\Scripts\activate
PS> python
Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import site
>>> site.getsitepackages()
['C:\\projects\\test', 'C:\\projects\\test\\lib\\site-packages']
```

**Expected:**

```console
['C:\\projects\\test\\lib\\site-packages']
```

**Environment**

- Operating system and architecture:: Windows 10 / amd64
- CPython versions tested on: 3.10.10 but also 3.10.4

### Linked PRs
* gh-102986

**Comment by ericvsmith** [comment]
Is it adding the current directory, or is it always `<venv>`? Try cd'ing somewhere else and see what it prints.

**Comment by timendum** [comment]
@ericvsmith It's always `<venv>`

```console
PS> cd c:\projects
PS> .\test\Scripts\python
Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import site
>>> site.getsitepackages()
['C:\\projects\\test', 'C:\\projects\\test\\lib\\site-packages']
>>> import os
>>> os.getcwd()
'C:\\projects'
```

The same also from other folders, outside `c:\projects`.

**Comment by zooba** [comment]
I'm really really hesitant to change any paths. People are always relying on them. (I've marked the PR to not be merged for the time being, just to make sure nobody jumps in.)

What problem is this causing? Given that modules in the path are importable, why shouldn't it be returned? If we stop returning it here, does it change the resulting `sys.path`?

(Yes, it sucks that Windows has different default `sys.path` from other platforms. Unfortunately, it's a big breaking change to change it. Maybe one day, but not in a little fix like this without massive fanfare.)

**Comment by timendum** [comment]
Hi,
I've reported this bug because ruff use the output of this function to exclude folders to be "linted".

https://github.com/charliermarsh/ruff-lsp/blob/349701d9e2bec0f9bf0d998959040743e1c2ce18/ruff_lsp/utils.py#L23

This function is the same for every OS.

**Comment by zooba** [comment]
Why is it an issue that ruff is not linting your Python install?

Are you storing your project in this directory? That's now how venvs are supposed to be used. It should be a subdirectory within your project.

**Comment by timendum** [comment]
I create a virtualenv folder to store my project, then ruff ask Python the list of the site paths, folder that will be not linted (because they are system folders) but Python returns the virtualenv folder itself in the list, so my files are not linted.

With the provided patch, in my case, Python returns only `lib\site-packages` folder.

**Comment by zooba** [comment]
Right, that's not the intended use of the virtual environment, so we're not going to change the behaviour for everyone. Try storing your project in its own folder, and put the virtual environment inside of that.
